json-org/src/main/java/com/fasterxml/jackson/datatype/jsonorg/JSONArrayDeserializer.java

```diff
@@ -66,7 +66,7 @@ public JSONArray deserialize(JsonParser p, DeserializationContext ctxt)
                 array.put(p.getNumberValue());
                 continue;
             case VALUE_NUMBER_FLOAT:
-                array.put(p.getNumberValue());
+                array.put(p.getDecimalValue());
```

Comment:
Only now realizing that this is actually behavioral change too. Hmmh.
So even if improvement, might break code if the exposed number changes from `Double` to `BigDecimal`?
Or maybe JSONP API encapsulates so this is not the case?

Since I already merged changes for JSR-353/JSONP will merge this too, just noting that we may get bug reports for 2.15.

Reply:
@pjfanning  Doh. Not JSONP/JSR-353. Still, similar concerns.

But quick question: would it make sense to instead call `p.getNumberValueExact()`? This would at least retain exact `Float`/`Double` for binary formats? Would change it to `BigDecimal` for JSON and other textual formats, which is probably good overall.
WDYT?

Reply:
if `p.getNumberValueExact()` works, then I'd be happy with that change - let me try a PR with that instead

Reply:
`b.add(name, p.getNumberValueExact());` doesn't compile
ie b.add does not support Number

I looked at the code in the json libs and they tend to use BigDecimal under the hood - so I think the risks of the jackson changes are low.

I think it is better to take the risk that someone will report an edge case than to add complicated logic using p.getNumberValueExact() and then lots of instanceof checks with casts.

Reply:
Ahh. Yes, I forgot the challenge wrt non-typed `Number`. I concur with your assessment.
Ideally we'd have more time for testing but TBH this module probably won't get all that much beta-testing compared to `jackson-databind` or `jackson-core`.